### PR TITLE
Refactored JoinClass into a handler function

### PR DESCRIPTION
**Pull request by samarthupadhyaya27**

### CODE REVIEW

File: handler/class_test.go [hunk 1]
```diff
@@ -396,3 +395,167 @@ func TestDeleteClass(t *testing.T) {
...
+		req := httptest.NewRequest(http.MethodPost, "/", strings.NewReader(`{"uid": "test", "cid": "test"}`))
+		rec := httptest.NewRecorder()
+		assert.NotNil(t, req, rec)
+		c := echo.New().NewContext(req, rec)
+
+		if assert.NoError(t, handler.JoinClass(&db.DBContext{
+			Context: c,
+			TLADB:   d,
+		})) {
+			require.Equal(t, http.StatusOK, rec.Code)
```
Comment (lumisphere902) [inline handler/class_test.go:418]: Small nit: check for something being returned (the class data)

**Review by lumisphere902 — APPROVED** [review]
Hey Samarth, this looks really good! :) Glad to see you're a lot more efficient with this type of PR now. I left a few comments, but if everything looks good to you feel free to merge.

File: handler/class.go [hunk 2]
```diff
@@ -150,3 +150,81 @@ func DeleteClass(cc echo.Context) error {
 
 	return c.String(http.StatusOK, "")
 }
+
+func addClassToUser(u *db.User, cid string) {
+	alreadyAssociated := false
+	for _, class := range (*u).Classes {
+		if class == cid {
+			alreadyAssociated = true
```
[the remark this answers is not in the document]
Comment (samarthupadhyaya27) [reply handler/class.go]: That was a bad decision. I've removed a lot of code from that function.

File: handler/class.go [hunk 3]
```diff
@@ -150,3 +150,81 @@ func DeleteClass(cc echo.Context) error {
...
+	if !alreadyAssociated {
+		(*u).Classes = append((*u).Classes, cid)
+	}
+}
+
+func addUserToClass(uid string, c *db.Class) {
+	alreadyAssociated := false
+	for _, user := range (*c).Members {
+		if user == uid {
+			alreadyAssociated = true
```
[the remark this answers is not in the document]
Comment (samarthupadhyaya27) [reply handler/class.go]: Same fix as above.

File: handler/class_test.go (hunk 1 above)
Comment (samarthupadhyaya27) [reply handler/class_test.go:418]: I added tests verifying the class data is returned

**Review by lumisphere902 — APPROVED** [review]
Re-approving; thanks for all the work! :)

**Pull request merged**